src/rust/runtime/queue/qdesc.rs

```diff
@@ -6,23 +6,37 @@
 //==============================================================================
 
 /// IO Queue Descriptor
-#[derive(From, Into, Debug, Eq, PartialEq, Hash, Copy, Clone)]
+#[derive(Debug, Eq, PartialEq, Hash, Copy, Clone)]
 pub struct QDesc(usize);
 
 //==============================================================================
 // Trait Implementations
 //==============================================================================
 
-/// Convert Trait Implementation for Signed 32-bit Integers
 impl From<QDesc> for i32 {
+    /// Converts a [QDesc] to a [i32].
     fn from(val: QDesc) -> Self {
         val.0 as i32
```

Comment:
On a 64-bit machine, this conversion will lose half of the QDesc's bits.

Reply:
Yes, you are right, and this existed before as well =( See my previous comment.

Reply:
Yes, this is directly related to the usize issue, so if you push that to a new Issue, this should get resolved there as well.

Reply:
Sure, I have created this issue: https://github.com/demikernel/demikernel/issues/430